src/examples/time-offset/time-offset.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+// This example demonstrates how to set the system time.
+//
+// Usually, boards don't keep time on power cycles and restarts, it resets to Unix epoch.
+// The system time can be set by calling runtime.AdjustTimeOffset().
+//
+// Possible time sources: an external RTC clock or network (WiFi access point or NTP server)
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+)
+
+const myTime = "2006-01-02T15:04:05Z" // this is an example time you source somewhere
+
+func main() {
+
+	t, _ := time.Parse(time.RFC3339, myTime)
+	runtime.AdjustTimeOffset(-1 * int64(time.Since(t)))
```

Comment:
I think this may better explain what is happening (I didn't test the code though):

```suggestion
	// measure how many nanoseconds the internal clock is behind
	timeOfMeasurement := time.Now()
	actualTime, _ := time.Parse(time.RFC3339, myTime)
	offset := actualTime.Sub(timeOfMeasurement)

	// adjust internal clock by adding the offset to the internal clock
	runtime.AdjustTimeOffset(int64(offset)
```

Reply:
The original idea with this example was to give people a simple recipe to set custom time, not necessarily educational.

The API is a bit confusing, to be fair. It would be much simpler if one could just `time.Parse().UnixNano()` and set that value and the internals figure out how exactly to do that.

But if you insist on educational value of the example, sure we can expand it and probably also write some comments how it looks internally?
Shall we?